### DISCUSSION TITLE
Update All Outdated Packages Script

### DISCUSSION THREAD

**Discussion by aagamezl** [ScoopInstaller/Scoop#5539] [Show and tell]
# Description

As part of my development setup automation repository, today I created this Powershell script to check what packages need to be updated and update all the packages automatically for you, but giving the option to process or not before performing the update.

I will try to create a more complete script maybe giving the option to the user to select what packages to update from the list, but I'm not a Powershell magician, so will take some time or maybe will never be developed, but right now is useful enough at least for me.

Please give me feedback and/or ideas for the script: [update-scoop-packages.ps1](https://github.com/aagamezl/dev-env-config/blob/master/scoop/update-scoop-packages.ps1)

# Usage

I put the script in my Windows user directory, so is easy to run `c:\Users\[USERNAME]\update-scoop-packages.ps1` and can be run it like this `.\update-scoop-packages.ps1`.

**Comment by HUMORCE** [comment]
`scoop update *`

**Reply by aagamezl** [reply]
@HUMORCE Yes, the problem is that update everything, I'm trying to get some power of decision over what packages update automatically; right now I have the multi-select option done, but testing a more advanced mechanism involving direction keys.

**Reply by HUMORCE** [reply]
You can decide which updates, when `scoop status` outputs list of outdated apps. for apps that you don't want to update at the time, you can use `scoop hold <app>` to skips them in executing `scoop update *`.

If you are referring to the interaction options, then there is no such design.

**Reply by aagamezl** [reply]
Yes, I know about that way to work with scoop, just wanted something more interactive without having to hold apps one by one in the case of not wanting to update the app one time.

I will try to update my script ASAP and ask for more feedback. Thank you so much for taking the time to talk about this idea.
